### PR TITLE
Smit/cancel command error

### PR DESCRIPTION
**Pull request by smitshah2531**
### What does this PR do?

Fixed "Cancelling command execution throws errors" 

### What issues does this PR fix or reference?
@W-5757070@

### Functionality Before For "Cancelling Command Execution Throws Error"

Before

### Functionality After For "Cancelling Command Execution Throws Error"

After
 
### Functionality Before For "Parsing Error" when running  "SFDX: Deploy Source to Org"

Before_Parsing

### Functionality After  For "Parsing Error" when running  "SFDX: Deploy Source to Org"

After_Parsing

### CODE REVIEW

File: packages/salesforcedx-vscode-core/src/notifications/notificationService.ts [hunk 1]
```diff
@@ -84,13 +84,12 @@ export class NotificationService {
     cancellationToken?: vscode.CancellationToken
   ) {
     observable.subscribe(async data => {
-      if (data !== undefined && data.toString() === '0') {
+      if (data !== undefined && data !== null && data.toString() === '0') {
```
Comment (xyc) [inline packages/salesforcedx-vscode-core/src/notifications/notificationService.ts:87]: To simply, you can just use `String(data)`

Reference: https://exploringjs.com/impatient-js/ch_strings.html#converting-to-string

File: packages/salesforcedx-vscode-core/src/notifications/notificationService.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ import { DEFAULT_SFDX_CHANNEL } from '../channels';
 import { STATUS_BAR_MSG_TIMEOUT_MS } from '../constants';
 import { nls } from '../messages';
 import { sfdxCoreSettings } from '../settings';
+// import { stringify } from 'querystring';
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/notifications/notificationService.ts:15]: let's remove this import if it's not being used

Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/notifications/notificationService.ts:15]: Sounds good. I wasn't sure should I keep it or not but sure I will update it.

File: packages/salesforcedx-vscode-core/src/notifications/notificationService.ts (hunk 1 above)
Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/notifications/notificationService.ts]: sounds good. Thanks!!!

File: packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts [hunk 3]
```diff
@@ -59,20 +59,22 @@ export abstract class BaseDeployExecutor extends SfdxCommandletExecutor<
...
-          channelService.showChannelOutput();
-          handleDiagnosticErrors(
-            errors,
-            workspacePath,
-            execFilePathOrPaths,
-            BaseDeployExecutor.errorCollection
-          );
-        } else {
-          BaseDeployExecutor.errorCollection.clear();
+        if (stdOut !== '') {
```
Comment (AnanyaJha) [inline packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts:62]: can we make this a positive check instead?
ie. `if (stdOut) { doSomething(); }`

Comment (smitshah2531) [reply packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts]: Yes, we can make it a positive check. Solved!!! Thanks!!!

**Pull request merged**